nautobot/extras/management/__init__.py

```diff
@@ -172,15 +172,15 @@ def create_custom_statuses(
             # has had its name, slug, color and/or description changed from the defaults.
             # First, try to find by slug
             defaults = choice_kwargs.copy()
-            slug = defaults.pop("slug")
+            name = defaults.pop("name")
             try:
                 # may fail if a status with a different slug has a name matching this one
-                obj, created = Status.objects.get_or_create(slug=slug, defaults=defaults)
+                obj, created = Status.objects.get_or_create(name=name, defaults=defaults)
             except IntegrityError:
-                # OK, what if we look up by name instead?
-                defaults = choice_kwargs.copy()
-                name = defaults.pop("name")
+                # Integrity error would occur when running extras.0004_populate_default_status_records migration
+                # since slug is still present at that point
                 try:
+                    defaults["slug"] = slugify(name)
                     obj, created = Status.objects.get_or_create(name=name, defaults=defaults)
                 except IntegrityError as err:
                     raise SystemExit(
```

Comment:
This change set makes me a bit uncomfortable. We *shouldn't* be running `create_custom_statuses` after your data model changes, it should *only* be run by earlier migrations. So are these changes actually even needed?